homeassistant/components/media_player/onkyo_serial.py

```diff
@@ -0,0 +1,198 @@
+    'https://github.com/rayzorben/onkyo-serial/archive/0.0.2.zip'
+    '#onkyo_serial>=0.0.2'
+]
+
+_LOGGER = logging.getLogger(__name__)
+
+SUPPORT_ONKYO = SUPPORT_VOLUME_SET | SUPPORT_VOLUME_MUTE | \
+    SUPPORT_TURN_ON | SUPPORT_TURN_OFF | SUPPORT_SELECT_SOURCE | SUPPORT_PLAY
+
+CONF_SOURCES = 'sources'
+CONF_ZONES = 'zones'
+CONF_PORT = 'port'
+
+DEFAULT_NAME = 'Onkyo Serial Receiver'
+DEFAULT_PORT = '/dev/ttyUSB0'
+DEFAULT_ZONES = load("""
+master:
+    commands:
+        power:  'PWR'
+        volume: 'MVL'
+        source: 'SLI'
+        mute:   'AMT'
+    queries:
+        power:  'PWRQSTN'
+        volume: 'MVLQSTN'
+        source: 'SLIQSTN'
+        mute:   'AMTQSTN'
+zone2:
+    commands:
+        power:  'ZPW'
+        volume: 'ZVL'
+        source: 'SLZ'
+        mute:   'ZMT'
+    queries:
+        power:  'ZPWQSTN'
+        volume: 'ZVLQSTN'
+        source: 'SLZQSTN'
+        mute:   'ZMTQSTN'
+""")
+
+ZONE_SETUP_SCHEMA = vol.Schema({
+    vol.Required('commands'):
+        vol.Schema({vol.All(cv.string): vol.Any(cv.string)}),
+    vol.Required('queries'):
+        vol.Schema({vol.All(cv.string): vol.Any(cv.string)})
+})
+ZONE_SCHEMA = vol.Schema({vol.All(cv.string): ZONE_SETUP_SCHEMA})
+
+SOURCES_SCHEMA = vol.Schema({vol.All(cv.string): vol.Any(cv.string)})
+
+PLATFORM_SCHEMA = PLATFORM_SCHEMA.extend({
+    vol.Required(CONF_PORT): DEFAULT_PORT,
+    vol.Optional(CONF_NAME, default=DEFAULT_NAME): cv.string,
+    vol.Optional(CONF_ZONES): ZONE_SCHEMA,
+    vol.Optional(CONF_SOURCES, SOURCES_SCHEMA):
+        {cv.string: cv.string},
+})
+
+
+def setup_platform(hass, config, add_devices, discovery_info=None):
+    """Setup the Onkyo platform."""
+    port = config.get(CONF_PORT)
+
+    # merge the default zones overridden with the configuration.yaml
+    # and create one device per zone
+    zones = DEFAULT_ZONES
+    if CONF_ZONES in config:
+        zones = dict(zones.items() + config.get(CONF_ZONES))
+
+    devices = list()
+    for key in zones.keys():
+        device = OnkyoSerialDevice(
+            zones,
+            key,
+            port=port,
+            sources=config.get(CONF_SOURCES),
+            name="Onkyo-{zone}".format(zone=key)
+        )
+        devices.append(device)
+
+    add_devices(devices)
+
+
+class OnkyoSerialDevice(MediaPlayerDevice):
+    """Representation of an Onkyo device."""
+
+    def __init__(self, config, zone, port=None, sources=None, name=None):
+        """Initialize the Onkyo Receiver."""
+        from onkyo_serial import OnkyoSerial
+        from onkyo_serial import ONKYO_SOURCES
+        self._zone = zone
+        self._mute = False
+        self._volume = 0
+        self._source = None
+        self._power = STATE_OFF
+        self._name = name
+        self._sources = ONKYO_SOURCES
+        if sources:
+            self._sources = dict(self._sources.items() + sources.items())
+        self._source_list = list(self._sources.values())
+        #pylint: disable=E

```

Comment:
this looks like a bug on houndci-bot, but the disable is because pylint complains that Instance of 'OnkyoSerial' has no 'on_state_change' member when it does and is working fine.